src/Data/Text/Utf8/AhoCorasick/Automaton.hs

```diff
@@ -304,10 +304,10 @@ asciiCount = 128
 -- O(1) lookup of a transition, rather than doing a linear scan over all
 -- transitions. The fallback goes back to the initial state, state 0.
 buildAsciiTransitionLookupTable :: IntMap State -> TypedByteArray Transition
-buildAsciiTransitionLookupTable transitions = TBA.fromList $ map (\i ->
+buildAsciiTransitionLookupTable transitions = TBA.fromList $ [0..asciiCount - 1] <&> \i ->
```

Comment:
IMO `map` was more readable :sweat_smile: 

Alternatively, writing a generate function for `TypedByteArray` that uses a mutable byte array and then `unsafeFreeze`s it could also be an idea, as that should work in the `ST` monad as well.

Reply:
I see that :D I just didn't like that the list was trailing after that closing paren a few lines later. I guess `flip map` would do the trick as well but I think I'll just take a few minutes to write a generate function :+1: Seems to be the best solution.